### PR TITLE
feat: Add data collection dialog and settings

### LINKED ISSUE #4766: [Telemetry] Add data collection dialog and the ability to change selection in settings
## Description
Composer should prompt the user with a dialog to opt-in or opt-out of data collection when the user first opens the app. The dialog should not reappear upon subsequent sessions, and it should not conflict with any of the PVA logic or onboarding flows. The user should also have the ability to change their selection in the application settings.

### PR DESCRIPTION
**Pull request by tdurnford**
## Description
Adds a prompt when the user opens the app to opt-in or opt-out of data collection. It also adds the ability for them to change their data collection preferences in the Application Settings. 

## Task Item
Closes #4766

### Data Collection Dialog
image

### Data Collection Application Settings
image

### CODE REVIEW

File: Composer/packages/client/src/recoilModel/dispatchers/serverSettings.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+      set(ServerSettingsState, settings);
+    } catch (error) {
+      logMessage(callbackHelpers, `Error fetching data collection settings: ${error}`);
+    }
+  });
+
+  return {
+    fetchServerSettings,
+    setAllowDataCollection,
```
Comment (hatpick) [inline Composer/packages/client/src/recoilModel/dispatchers/serverSettings.tsx:49]: per server setting in the future a new dispatcher callback is required, have you considered having one callback to update settings like:
```
const updateServerSettings= useRecoilCallback(
    (callbackHelpers: CallbackInterface) => async (settings: Partial<ServerSettings>) => {
        // merge with current settings and update app and server state
    }
  )
```

File: Composer/packages/types/src/telemetry.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+export type TelemetrySettings = {
+  allowDataCollection?: boolean | null;
```
[the remark this answers is not in the document]
Comment (tdurnford) [reply Composer/packages/types/src/telemetry.ts:5]: It's not necessary - I just need to change the default value to `false` in the app state.

File: Composer/packages/client/src/recoilModel/dispatchers/serverSettings.tsx (hunk 1 above)
Comment (tdurnford) [reply Composer/packages/client/src/recoilModel/dispatchers/serverSettings.tsx:49]: Sounds good to me

File: Composer/packages/types/src/telemetry.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+export type TelemetrySettings = {
+  allowDataCollection: boolean | null;
```
Comment (hatpick) [inline Composer/packages/types/src/telemetry.ts:5]: I still think this should be:
{
   allowDataCollection ?: boolean;
}

**Review by hatpick — CHANGES_REQUESTED** [review]

**Review by hatpick — APPROVED** [review]

**Pull request merged**